### PR TITLE
scylla_node: start: add replace_node parameter

### PR DESCRIPTION
**Pull request by bhalevy**
To be used for replacing a node given its host_id
rather than its ip address.

Makes use of the newly added --replace-node-first-boot scylla option.

Signed-off-by: Benny Halevy <bhalevy@scylladb.com>

### CODE REVIEW

**Comment by bhalevy** [comment]
Ref https://github.com/scylladb/scylladb/pull/12316

This change does not directly depend on the option being added since no code currently
uses the new `replace_node` parameter, while the change is backward compatible with `replace_address`.
but we should wait with merging it until the scylla PR is finalized and merged.

File: ccmlib/scylla_node.py [hunk 1]
```diff
@@ -563,7 +569,10 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
             args += ['--overprovisioned']
         if '--prometheus-address' not in args:
             args += ['--prometheus-address', data['api_address']]
-        if replace_address:
+        if replace_node:
+            assert replace_address is None, "replace_node and replace_address cannot be specified together"
+            args += ['--replace-node-first-boot', replace_node]
```
Comment (fruch) [inline ccmlib/scylla_node.py:574]: what doesn't first-boot mean, would anyone thing to call it in a different context ? i.e. passing it not on first boot ? 

and it's a bit weird that this commit doesn't say host_id anywhere... i.e. `replace_node` is actually `replace_node_host_id`, i.e. nothing here indicate what's need to be passing down.

**Comment by bhalevy** [comment]
https://github.com/scylladb/scylladb/pull/12316 is merged

File: ccmlib/scylla_node.py (hunk 1 above)
Comment (bhalevy) [reply ccmlib/scylla_node.py:574]: It basically follows the convention of --replace-endpoint-first-boot.
The details are in the respective documentation that was updated in https://github.com/scylladb/scylladb/pull/12316.

**Comment by fruch** [comment]
> [scylladb/scylladb#12316](https://github.com/scylladb/scylladb/pull/12316) is merged

@bhalevy this have conflict with other change you did with the boot sequence logs

**Comment by bhalevy** [comment]
> > [scylladb/scylladb#12316](https://github.com/scylladb/scylladb/pull/12316) is merged
> 
> @bhalevy this have conflict with other change you did with the boot sequence logs

rebasing...

**Comment by bhalevy** [comment]
@fruch branch was rebased

**Review by fruch — CHANGES_REQUESTED** [review]
LGTM

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**